### PR TITLE
update commander to show empty array as default value

### LINKED ISSUE #3433: Empty default value in cli options
```
--globals <names>                       allow the given comma-delimited global [names] (default: )
```

If default value is an empty list(`[]`), it shows as `(default: )`. It is confusing to users. No default or `(default: [])` is better.

### PR DESCRIPTION
**Pull request by outsideris**
### Description of the Change
In `mocha`'s help, it shows nothing if the default value is an empty array. It is confusing for users.
```console
--globals <names>                       allow the given comma-delimited global [names] (default: )
```

[commander v2.19.0](https://github.com/tj/commander.js/releases/tag/v2.19.0) fixed the display of falsy default values in its `--help` output.

This change updates our package dependency to this version of `tj/commander.js`.

### Alternate Designs
N/A

### Why should this be in core?
`default: ` without a value is rather meaningless.

### Benefits
Now, it will show:
```console
--globals <names>                       allow the given comma-delimited global [names] (default: [])
```

### Applicable issues
Fixes #3433
See [tj/commander.js/#855](https://github.com/tj/commander.js/issues/855)

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -779,73 +779,71 @@ Mocha supports the `err.expected` and `err.actual` properties of any thrown `Ass
...
-    --forbid-pending                        causes pending tests and test marked with skip to fail the suite
-    --file <file>                           include a file to be ran during the suite (default: )
-    --exclude <file>                        a file or glob pattern to ignore (default: )
-    -h, --help                              output usage information
-
-  Commands:
-
-    init <path>  initialize a client-side mocha setup at <path>
+```console
+Usage: mocha [debug] [options] [files]
```
Comment (plroebuck) [inline docs/index.md]: Hadn't noticed this before, but where did that "[debug]" part come from?

File: docs/index.md [hunk 2]
```diff
@@ -779,73 +779,71 @@ Mocha supports the `err.expected` and `err.actual` properties of any thrown `Ass
...
+  -R, --reporter <name>                   specify the reporter to use (default: "spec")
+  -S, --sort                              sort test files
+  -b, --bail                              bail after first test failure
+  -d, --debug                             enable node's debugger, synonym for node --debug
+  -g, --grep <pattern>                    only run tests matching <pattern>
+  -f, --fgrep <string>                    only run tests containing <string>
+  -gc, --expose-gc                        expose gc extension
+  -i, --invert                            inverts --grep and --fgrep matches
+  -r, --require <name>                    require the given module
+  -s, --slow <ms>                         "slow" test threshold in milliseconds [75]
```
Comment (plroebuck) [inline docs/index.md]: shouldn't this say `(default: 75)`?

File: docs/index.md [hunk 3]
```diff
@@ -779,73 +779,71 @@ Mocha supports the `err.expected` and `err.actual` properties of any thrown `Ass
...
+  -t, --timeout <ms>                      set test-case timeout in milliseconds [2000]
```
Comment (plroebuck) [inline docs/index.md]: shouldn't this say `(default: 2000)`?

File: docs/index.md [hunk 4]
```diff
@@ -779,73 +779,71 @@ Mocha supports the `err.expected` and `err.actual` properties of any thrown `Ass
...
+  -O, --reporter-options <k=v,k2=v2,...>  reporter-specific options
+  -R, --reporter <name>                   specify the reporter to use (default: "spec")
+  -S, --sort                              sort test files
+  -b, --bail                              bail after first test failure
+  -d, --debug                             enable node's debugger, synonym for node --debug
+  -g, --grep <pattern>                    only run tests matching <pattern>
+  -f, --fgrep <string>                    only run tests containing <string>
+  -gc, --expose-gc                        expose gc extension
+  -i, --invert                            inverts --grep and --fgrep matches
+  -r, --require <name>                    require the given module
```
Comment (plroebuck) [inline docs/index.md]: shouldn't this say `(default: [])`?

File: docs/index.md [hunk 5]
```diff
@@ -779,73 +779,71 @@ Mocha supports the `err.expected` and `err.actual` properties of any thrown `Ass
...
+  --no-timeouts                           disables timeouts, given implicitly with --debug
+  --no-warnings                           silence all node process warnings
+  --opts <path>                           specify opts path (default: "test/mocha.opts")
+  --perf-basic-prof                       enable perf linux profiler (basic support)
+  --napi-modules                          enable experimental NAPI modules
+  --prof                                  log statistical profiling information
+  --log-timer-events                      Time events including external callbacks
+  --recursive                             include sub directories
+  --reporters                             display available reporters
+  --retries <times>                       set numbers of time to retry a failed test case
```
Comment (plroebuck) [inline docs/index.md]: shouldn't this say `(default: 0)`?

File: docs/index.md [hunk 6]
```diff
@@ -779,73 +779,71 @@ Mocha supports the `err.expected` and `err.actual` properties of any thrown `Ass
...
+  --trace                                 trace function calls
+  --trace-deprecation                     show stack traces on deprecations
+  --trace-warnings                        show stack traces on node process warnings
+  --use_strict                            enforce strict mode
+  --watch-extensions <ext>,...            specify extensions to monitor with --watch (default: ["js"])
+  --delay                                 wait for async suite definition
+  --allow-uncaught                        enable uncaught errors to propagate
+  --forbid-only                           causes test marked with only to fail the suite
+  --forbid-pending                        causes pending tests and test marked with skip to fail the suite
+  --file <file>                           include a file to be ran during the suite (default: [])
```
Comment (plroebuck) [inline docs/index.md]: think this description should be:
```console
  --file <file>  adds file be loaded prior to suite execution (default: [])
```

File: docs/index.md [hunk 7]
```diff
@@ -779,73 +779,71 @@ Mocha supports the `err.expected` and `err.actual` properties of any thrown `Ass
...
+  --exclude <file>                        a file or glob pattern to ignore (default: [])
```
Comment (plroebuck) [inline docs/index.md]: think this description should be:
```console
  --exclude <file>  adds file or glob pattern to ignore (default: [])
```

**Comment by outsideris** [comment]
fixed.

File: bin/_mocha [hunk 8]
```diff
@@ -157,9 +157,9 @@ program
   .option('-f, --fgrep <string>', 'only run tests containing <string>')
   .option('-gc, --expose-gc', 'expose gc extension')
   .option('-i, --invert', 'inverts --grep and --fgrep matches')
-  .option('-r, --require <name>', 'require the given module')
-  .option('-s, --slow <ms>', '"slow" test threshold in milliseconds [75]')
-  .option('-t, --timeout <ms>', 'set test-case timeout in milliseconds [2000]')
+  .option('-r, --require <name>', 'require the given module', [])
```
Comment (plroebuck) [inline bin/_mocha]: Was this verified with two or more? Why wasn't `collect` needed?

File: bin/_mocha [hunk 9]
```diff
@@ -221,7 +221,8 @@ program
   .option('--reporters', 'display available reporters')
   .option(
     '--retries <times>',
-    'set numbers of time to retry a failed test case'
+    'set numbers of time to retry a failed test case',
```
Comment (plroebuck) [inline bin/_mocha]:
```console
'set number of times to retry a failed test case',
```

File: bin/_mocha [hunk 10]
```diff
@@ -221,7 +221,8 @@ program
   .option('--reporters', 'display available reporters')
```
Comment (plroebuck) [inline bin/_mocha]:
```js
  .option('--reporters', 'display available reporters and exit')
```

File: bin/_mocha [hunk 11]
```diff
@@ -157,9 +157,9 @@ program
   .option('-f, --fgrep <string>', 'only run tests containing <string>')
   .option('-gc, --expose-gc', 'expose gc extension')
   .option('-i, --invert', 'inverts --grep and --fgrep matches')
-  .option('-r, --require <name>', 'require the given module')
-  .option('-s, --slow <ms>', '"slow" test threshold in milliseconds [75]')
-  .option('-t, --timeout <ms>', 'set test-case timeout in milliseconds [2000]')
+  .option('-r, --require <name>', 'require the given module', [])
+  .option('-s, --slow <ms>', '"slow" test threshold in milliseconds', 75)
+  .option('-t, --timeout <ms>', 'set test-case timeout in milliseconds', 2000)
```
Comment (plroebuck) [inline bin/_mocha]:
```js
  .option('-t, --timeout <ms>', 'specify test timeout threshold in milliseconds', 2000)
```

File: bin/_mocha [hunk 12]
```diff
@@ -157,9 +157,9 @@ program
   .option('-f, --fgrep <string>', 'only run tests containing <string>')
   .option('-gc, --expose-gc', 'expose gc extension')
   .option('-i, --invert', 'inverts --grep and --fgrep matches')
-  .option('-r, --require <name>', 'require the given module')
-  .option('-s, --slow <ms>', '"slow" test threshold in milliseconds [75]')
-  .option('-t, --timeout <ms>', 'set test-case timeout in milliseconds [2000]')
+  .option('-r, --require <name>', 'require the given module', [])
+  .option('-s, --slow <ms>', '"slow" test threshold in milliseconds', 75)
```
Comment (plroebuck) [inline bin/_mocha]:
```js
  .option('-s, --slow <ms>', 'specify "slow" test threshold in milliseconds', 75)
```

**Comment by outsideris** [comment]
Fixed.

**Pull request merged**